lib/events/emitter.go

```diff
@@ -300,12 +300,14 @@ func NewLoggingEmitter() *LoggingEmitter {
 type LoggingEmitter struct {
 }
 
-// EmitAuditEvent logs audit event, skips session print events
-// and session disk events, because they are very verbose
+// EmitAuditEvent logs audit event, skips session print events, session
+// disk events and app session request events, because they are very verbose.
 func (*LoggingEmitter) EmitAuditEvent(ctx context.Context, event AuditEvent) error {
 	switch event.GetType() {
 	case ResizeEvent, SessionDiskEvent, SessionPrintEvent, "":
 		return nil
+	case AppSessionRequestCode:
+		return nil
```

Comment:
I'm not exactly sure what's intended here but why is this limited to `LoggingEmitter`? Shouldn't this event be dropped either with all emitters or none? At the very least I suppose it should be added to

https://github.com/gravitational/teleport/blob/e096e5608fea468b828657f0e6b591cec02be6ef/lib/events/emitter.go#L561-L566

Reply:
The idea was to make the `app.session.request` not valid event in AuditLog  thus only AditLog emitter was aligned. 
Under the hood the TeeStream.EmitEvent will delegate the EmitEvent call to AuditLog.EmitEvent handled where `AppSessionRequestCode` is skipped, but I will add it here for consistency.

Reply:
> as the issue was related to sync recording modes only?

I was able to reproduce the issue with async app stream mode. 
The issue is related to how authService handles stream upload not how app service uploads the events to auth service.

Is async recoding mode events are dispatched only to s.streamer  where chunks of events are uploaded to the appropriate backend. 

Whereas in sync mode events all also dispatched to  a.emitter (MultiLog combined with { `LoggingEmitter` and `AuditLog`}) that leads to emitting app.session.request event in logs and audit log storage.  

https://github.com/gravitational/teleport/blob/555695dfdd88f3e8222d52e5739b45775c0c6d12/lib/auth/auth.go#L2127
```go
// modeStreamer creates streamer based on the event mode
func (a *Server) modeStreamer(ctx context.Context) (events.Streamer, error) {
	recConfig, err := a.GetSessionRecordingConfig(ctx)
	if err != nil {
		return nil, trace.Wrap(err)
	}
	// In sync mode, auth server forwards session control to the event log
	// in addition to sending them and data events to the record storage.
	if services.IsRecordSync(recConfig.GetMode()) {
		return events.NewTeeStreamer(a.streamer, a.emitter), nil
	}
	// In async mode, clients submit session control events
	// during the session in addition to writing a local
	// session recording to be uploaded at the end of the session,
	// so forwarding events here will result in duplicate events.
	return a.streamer, nil
}
```

> shouldn't the event be dropped more comprehensively involving the other kinds of emitters as well (incl. file-based events logs, dynamo, firestore, ...)?

what about current setup where we wave events that are dropped only in `LoggingEmitter` and `AuditLog ` like `ResizeEvent`, `SessionDiskEvent` and `SessionPrintEvent`  ? 

Reported issue scope was only related to AuditLogs. If we drop only  app.session.request for all emitters this approach won't be consistent with current `SessionPrintEvent` `ResizeEvent` and `SessionPrintEvent`  event handling.

Reply:
> I was able to reproduce the issue with async app stream mode.

What do you mean by this? Have you been able to observe `app.session.request`-related "Unknown"s in the proxy web UI even with an async recording mode (i.e. plain `node` or `proxy`)?

As you explained, since the individual emitters (file-based, dynamo, firestore, etc.) are wrapped in a `TeeStreamer` anyway, the presented solution (adding an `AppSessionRequestEvent` case to `LoggingEmitter` and `TeeStream`) should be sufficient. In that case I'd however revert the changes made to `lib/srv/app/session.go`.

Reply:
>> I was able to reproduce the issue with async app stream mode.

> What do you mean by this? Have you been able to observe app.session.request-related "Unknown"s in the proxy web UI even with an async recording mode (i.e. plain node or proxy)?

Sorry, that was unclear, by "async app stream mode" I meant the way how app service creates steam. If async stream was enforced by deletion of  sync condition from [code](https://github.com/gravitational/teleport/blob/master/lib/srv/app/session.go#L176) the problem still exists in `session_recording: "node-sync"` or `proxy-sync`  when AppSessionRequestEvent is not skipped from AuditLog. 

So as you pointed changes made in `lib/srv/app/session.go` (enforcing async emitter) are superfluous  and can be removed. 

> In that case I'd however revert the changes made to lib/srv/app/session.go.

reverted.